### PR TITLE
Glossary: additional publication frequencies, incl. item_order

### PR DESCRIPTION
**Pull request by effjot**
Now all publication frequency terms from ISO 19115 are implemented and translated to German:
https://www.geoportal-bw.de/documents/20147/754293/ISO_19115_Deutsch_Stand_08-12-2008.pdf
table „B.5.18 MD_MaintenanceFrequencyCode“

I’ve set the version to 1.2.0 and introduced a file `upgrade_to_1.2.0.sql`.

### CODE REVIEW

File: pg_metadata/metadata.txt
Comment (Gustry) [inline pg_metadata/metadata.txt:9]: Same comment, you would need to revert this change. It's done automatically by the CI with https://github.com/opengisch/qgis-plugin-ci

File: pg_metadata/install/sql/pgmetadata/90_GLOSSARY.sql
Comment (Gustry) [inline pg_metadata/install/sql/pgmetadata/90_GLOSSARY.sql:41]: I'm not German so I will let you manage this, but you haven't capitalized any German label (L41, L42 etc).
Up to you to check these ;-)

**Review by Gustry — APPROVED** [review]

File: pg_metadata/metadata.txt
Comment (effjot) [reply pg_metadata/metadata.txt:9]: Thanks for the info!

File: pg_metadata/install/sql/pgmetadata/90_GLOSSARY.sql
Comment (effjot) [reply pg_metadata/install/sql/pgmetadata/90_GLOSSARY.sql:41]: That’s a good point. I used capital/small letters according to grammar. But I’ve just checked how QGIS handles this and found out that the first word in UI elements is always capitalised.
I think I’ll prepare a separate PR for the whole glossary with correct capitalisation.

**Comment by Gustry** [comment]
> By the way, is there way to use the item order in the attribute forms?

I think there is a checkbox "Order by value" in the field form settings.

**Comment by effjot** [comment]
> > By the way, is there way to use the item order in the attribute forms?
> 
> I think there is a checkbox "Order by value" in the field form settings.

That would sort alphabetically according to the  label_xx field (i.e. the values shown in the dropdown list). Unless it is well hidden, there doesn’t seem to be an option for a separate sort key.

**Comment by effjot** [comment]
I have changed the glossary terms introduced in this PR to begin with upper case letters.

@Gustry Should I extend `upgrade_to_1.2.0.sql` and `90_Glossary.sql` so that the existing terms (introduced before this PR) are corrected, too? Or would you prefer this in a separate PR?

**Comment by effjot** [comment]
@Gustry Github says “This branch has conflicts that must be resolved” for `upgrade_to_1.2.0.sql`. I think I need to rebase on upstream/master, correct?

Also, should I change case for existing terms (introduced before this PR) nere, too?

**Comment by effjot** [comment]
While we’re working on this PR, I’ve found 2 more additions to the glossary lying around in our local installation:
* file/link type for a directory (We use it to point to a folder on our fileserver containing additional documentation, e.g. legends, descriptions of data by the provider.)
* file/link type for ESRI Shapefiles (still very common distribution format; we store a link to the original download source)

````sql
INSERT INTO pgmetadata.glossary (field, code, label_en, description_en, item_order, label_fr, description_fr, label_it, description_it, label_es, description_es, label_de, description_de) VALUES ('link.type', 'directory', 'A directory', 'Directory on the local filesystem', NULL, NULL, NULL, NULL, NULL, NULL, NULL, 'Ein Ordner', 'Ein Ordner auf dem lokalen Dateisystem');
INSERT INTO pgmetadata.glossary (field, code, label_en, description_en) VALUES ('link.mime', 'directory', 'inode/directory', 'Directory (not an official MIME type');
INSERT INTO pgmetadata.glossary (field, code, label_en, description_en, item_order, label_fr, description_fr, label_it, description_it, label_es, description_es, label_de, description_de) VALUES ('link.type', 'ESRI:SHP', 'ESRI Shapefile', 'Vector layer in Shapefile format (.shp)', NULL, NULL, NULL, NULL, NULL, NULL, NULL, 'ESRI Shapefile', 'Vektorlayer im Shapefile-Format (.shp)');
````

**Comment by effjot** [comment]
> While we’re working on this PR, I’ve found 2 more additions to the glossary lying around in our local installation:
>     * file/link type for a directory (We use it to point to a folder on our fileserver containing additional documentation, e.g. legends, descriptions of data by the provider.)
>     * file/link type for ESRI Shapefiles (still very common distribution format; we store a link to the original download source)

Should I add these to the PR?

**Comment by effjot** [comment]
I’ve rebased the glossary additions in the original PR onto version 1.2.1 and added the suggestions for 2 new link types in a new commit.

**Comment by effjot** [comment]
Is there any work for me to do on this PR?
Thanks!

**Comment by effjot** [comment]
I have extended this PR to include @mixedbredie’s suggestions from #152 (with German translations from the INSPIRE website) and the “unknown“ terms I have proposed in #159.
Also, there are some more roles and frequencies we’ve encountered and would suggest as an addition to the glossary.

@Gustry and @mdouchin: could this PR be merged?
Thank you!

**Comment by effjot** [comment]
Just a little ping if this PR could be merged. ;-)
@Gustry @mdouchin

**Comment by Gustry** [comment]
I will have a look next week, very sorry for the delay !

**Comment by effjot** [comment]
Thank you!

**Comment by effjot** [comment]
Could you decide whether to merge this PR? Thanks!

**Review by mdouchin — CHANGES_REQUESTED** [review]
Sorry for the huge delay

**Review by Gustry — APPROVED** [review]
Hi, sorry again for the delay !

I approve for when the migration will be fixed (comment from my colleague)

File: pg_metadata/install/sql/upgrade/upgrade_to_1.2.0.sql
Comment (effjot) [reply pg_metadata/install/sql/upgrade/upgrade_to_1.2.0.sql:1]: The upgrade files for 1.2.0 and 1.2.2 are already in master. My understanding is that these are incremental, so that upgrades from any version to the current version are possible. However, you should know best. I will not touch them in this PR, to not further delay things, and you could remove them after merging the PR if necessary.

File: pg_metadata/install/sql/upgrade/upgrade_to_1.2.0.sql
Comment (Gustry) [inline pg_metadata/install/sql/upgrade/upgrade_to_1.2.0.sql:453]:
```suggestion
```

As version 1.2.0 is already out, this should be removed.
It's isolated in a commit in case you need it.

Even for CI, it's make sure that the migration 1.2.3 is self contained. (and not using this migration 1.2.0)

Comment (effjot) [reply pg_metadata/install/sql/upgrade/upgrade_to_1.2.0.sql:453]: Should I move those changes into the upgrade_1.3.0 im preparing now? Otherwise, users who are on a version before 1.2.0 and are going to update to 1.3.0 might miss those changes. (See file processing/upgrade.py, lines 154–170 `for sf in sql_files:`)

Comment (Gustry) [reply pg_metadata/install/sql/upgrade/upgrade_to_1.2.0.sql:453]: Yes, all changes to the database must go in a migration which is not released yet, that's why everything must go in a migration greater than 1.2.0.

https://github.com/3liz/qgis-pgmetadata-plugin/blob/master/pg_metadata/processing/database/upgrade.py#L151C42-L151C60 <= The current version of the database is used, so we have a list of migrations to apply. Someone having database version 1.0.0 for instance will launch, in the correct order, 3 migrations according to `available_migrations` : https://github.com/3liz/qgis-pgmetadata-plugin/tree/master/pg_metadata/install/sql/upgrade

Comment (effjot) [reply pg_metadata/install/sql/upgrade/upgrade_to_1.2.0.sql:453]: But haven’t version 1.2.0 – 1.2.2 already been released? (https://docs.3liz.org/qgis-pgmetadata-plugin/changelog/)

Comment (effjot) [reply pg_metadata/install/sql/upgrade/upgrade_to_1.2.0.sql:453]: I have renamed the former upgrade_1.2.3.sql to 1.3.0 and left 1.2.0 and 1.2.2 (no sql for 1.2.1) untouched, so I guess we’re on the same page now.
After you have merged this PR, I can update my PR #139.

Comment (Gustry) [reply pg_metadata/install/sql/upgrade/upgrade_to_1.2.0.sql:453]: Sorry, there is still the file 1.2.0 modified here.
Look at your PR : https://github.com/3liz/qgis-pgmetadata-plugin/pull/103/files

Comment (effjot) [reply pg_metadata/install/sql/upgrade/upgrade_to_1.2.0.sql:453]: Ok, now I get it. I always thought you wanted me to remove the *whole* file, which confused me because it already was released. Now I realised that my changes have been applied to the released version of the file because the PR is so extremely old. (Originally, the PR would have introduced the file in the first place, as I have now seen in the very first message here…)
I’ll clean it up tomorrow and then the PR will hopefully be up to date.

Comment (Gustry) [reply pg_metadata/install/sql/upgrade/upgrade_to_1.2.0.sql:453]: Indeed, we do not want to remove the old migration 🙂
Just edit a new migration file which has been released yet.

Thanks :)

**Review by Gustry — APPROVED** [review]
Thanks !

**Comment by effjot** [comment]
Great, thanks!

**Comment by effjot** [comment]
Maybe you could add French translations for the new glossary terms before releasing the new version? I can’t speak French and I dare not use Google translate. ;-)

**Comment by Gustry** [comment]
Yes, I was planning to do that before releasing the new version.
I guess you want to update/rebase #85 so we avoid having conflict together ;-)

**Pull request merged**
